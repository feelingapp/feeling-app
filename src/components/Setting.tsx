import Link from "next/link"
import { styled } from "linaria/react"

const Container = styled.a``

const Title = styled.div`
  font-size: 1.8rem;
  font-weight: 500;
`

const Subtitle = styled.div`
  color: var(--grey-color);
  font-size: 1.4rem;
`

interface Props {
  title: string
  subtitle?: string
  href?: string
}

export default function Setting(props: Props): JSX.Element {
  const { title, subtitle, href } = props

  return (
    <Container href={href}>
      <Title>{title}</Title>
      {subtitle && <Subtitle>{subtitle}</Subtitle>}
    </Container>
  )
}
