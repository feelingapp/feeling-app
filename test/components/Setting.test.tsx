import { shallow } from "enzyme"
import Setting from "../../src/components/Setting"

describe("Setting: ", () => {
  it("matches snapshot", () => {
    const wrapper = shallow(
      <Setting title="Lorem ipsum" subtitle="dolor sit amet" href="/" />
    )
    expect(wrapper).toMatchSnapshot()
  })
})
