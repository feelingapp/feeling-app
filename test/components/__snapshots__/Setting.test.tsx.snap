// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Setting:  matches snapshot 1`] = `
<a
  href="/"
>
  <Title>
    Lorem ipsum
  </Title>
  <Subtitle>
    dolor sit amet
  </Subtitle>
</a>
`;
